databuilder/extractor/dashboard/tableau/tableau_dashboard_extractor.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import requests
+import html
+import re
+
+from pyhocon import ConfigTree, ConfigFactory  # noqa: F401
+from typing import Any  # noqa: F401
+
+from databuilder import Scoped
+
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.extractor.dashboard.tableau.tableau_dashboard_utils import TableauDashboardUtils, TableauDashboardAuth, TableauGraphQLApiExtractor
+from databuilder.extractor.restapi.rest_api_extractor import STATIC_RECORD_DICT
+
+from databuilder.extractor.dashboard.tableau.tableau_dashboard_constants import *
+
+from databuilder.rest_api.rest_api_query import RestApiQuery  # noqa: F401
+from databuilder.rest_api.base_rest_api_query import BaseRestApiQuery  # noqa: F401
+from databuilder.rest_api.base_rest_api_query import EmptyRestApiQuerySeed
+from databuilder.rest_api.base_rest_api_query import RestApiQuerySeed
+from databuilder.rest_api.tableau.tableau_paginated_rest_api_query import TableauPaginatedRestApiQuery
+
+from databuilder.transformer.base_transformer import ChainedTransformer
+from databuilder.transformer.dict_to_model import DictToModel, MODEL_CLASS
+from databuilder.transformer.template_variable_substitution_transformer import \
+    TemplateVariableSubstitutionTransformer, TEMPLATE, FIELD_NAME as VAR_FIELD_NAME
+from databuilder.transformer.timestamp_string_to_epoch import TimestampStringToEpoch, FIELD_NAME
+
+LOGGER = logging.getLogger(__name__)
+
+
+class TableauDashboardExtractor(Extractor):
+    """
+    """
+
+    def init(self, conf):
+        # type: (ConfigTree) -> None
+
+        self._conf = conf
+        self._auth = TableauDashboardAuth(self._conf)
+        self.query = """query {
+            workbooks {
+                id
+                name
+                createdAt
+                description
+                projectName
+            }
+        }"""
+
+        self._extractor = self._build_extractor()
+
+        transformers = []
+        timestamp_str_to_epoch_transformer = TimestampStringToEpoch()
+        timestamp_str_to_epoch_transformer.init(
+            conf=Scoped.get_scoped_conf(self._conf, timestamp_str_to_epoch_transformer.get_scope()).with_fallback(
+                ConfigFactory.from_dict({FIELD_NAME: 'created_timestamp', })))
+        transformers.append(timestamp_str_to_epoch_transformer)
```

Comment:
I did this to follow the convention set by the Mode dashboard extractor, but I also feel that it might be easier to simply do it at extract time. +1 to discussing next time we sync